server/apiserver/accesslog/result_capturing_writer.go

```diff
@@ -0,0 +1,28 @@
+package accesslog
+
+import (
+	"net/http"
+)
+
+// resultCapturingWriter captures the size and status code of the response as http.response implements http.Flusher,
```

Comment:
// resultCapturingWriter captures the size and status code of the respons.
// Because http.response implements http.Flusher, we must do so too, otherwise `Watch*` methods don't work.
// We do not implement http.Hijacker, as HTTP/2 requests should not allow it.